src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -37,6 +38,37 @@ public T WithHttpClientFactory(IMsalHttpClientFactory httpClientFactory)
             return (T)this;
         }
 
+        /// <summary>
+        /// Allows developers to configure details about the authorities used, bypassing the need for MSAL
+        /// to perform instance discovery, e.g. https://aka.ms/aad-instance-discovery
+        /// MSAL uses the aliases to identify tokens that were issued to equivalent authorities, as tokens issued for one authority in the alias set is 
+        /// valid for all other aliases. For more details see https://aka.ms/msal-net-custom-instance-metadata
+        /// </summary>
+        /// <remarks>
+        /// Don't use this method unless you have specific needs that MSAL.NET's default behavior (fetching the instance metadata from the Azure AD discovery endpoint) does not fulfill.
+        /// Has no effect on ADFS or B2C authorities, only for AAD authorities</remarks>
+        /// <param name="instanceDiscoveryJson"></param>
```

Comment:
Per question above: Should clarify that when using this the developer takes the responsibility of this document holding critical information on valid authorities and this can be a security concern!

Note: This functionality needs to be properly documented in the thread model and highlighted in the next review

Reply:
Yeah, I thought about combining this with authority validation. IMO these are separate issues. Authority validation means: 
- [AAD] an instance discovery call, which MSAL performs anyway
- [B2C] not defined
- [ADFS] something ADFS specific

IMO we should deprecate the "validateAuthority" flag from all `WithAuthority()` builder methods, except for `WithAdfsAuthority()`. It makes no sense to have an authority validation flag on AAD since we perform validation anyway. 

Allowing the user to provide "valid authorities" does indeed bypass the `validate authority` mechanism.

Reply:
Updated comments.